src/panels/lovelace/components/hui-generic-entity-row.ts

```diff
@@ -146,7 +147,16 @@ export class HuiGenericEntityRow extends LitElement {
                                           100
                                       )}
                                       %`
-                                    : nothing)}
+                                    : this.config.secondary_info === "state"
+                                      ? html`${computeStateDisplay(
+                                          this.hass.localize,
+                                          stateObj,
+                                          this.hass.locale,
+                                          [],
+                                          this.hass.config,
+                                          this.hass.entities
+                                        )}`
```

Comment:
You should use `this.hass.formatEntityState`. Then you don't need to worry about numeric classes